### PR TITLE
[refactor](simd_json_reader) refactor simd json reader to adapt to parse multi json

### PR DESCRIPTION
**Pull request by sollhui**
## Proposed changes

The original json parsing method believed that a buffer only had one piece of data, which would cause loss of loaded data. 

The purpose of refactoring this module is to adapt to parse multiple json for one buffer. This is the pre pr of https://github.com/apache/doris/pull/27145

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by freemandealer** [comment]
@eldenmoon would you please take a look at this PR?

File: be/src/vec/exec/format/json/new_json_reader.cpp [hunk 1]
```diff
@@ -1437,48 +1459,38 @@ Status NewJsonReader::_append_error_msg(simdjson::ondemand::object* obj, std::st
...
+    RETURN_IF_ERROR(_init_json_parser_iterate(size, error));
     return Status::OK();
 }
-Status NewJsonReader::_simdjson_parse_json_doc(size_t* size, bool* eof) {
-    // read a whole message
-    SCOPED_TIMER(_file_read_timer);
-    const uint8_t* json_str = nullptr;
-    std::unique_ptr<uint8_t[]> json_str_ptr;
+
+Status NewJsonReader::_read_pipe_buf(size_t* size, bool* eof) {
```
Comment (freemandealer) [inline be/src/vec/exec/format/json/new_json_reader.cpp:1476]: the function name `read_pipe_buf` is not suitable here for the data source is not necessarily a pipe.

**Comment by HHoflittlefish777** [comment]
run buildall

**Review by eldenmoon — DISMISSED** [review]
LGTM

**Comment by HHoflittlefish777** [comment]
run buildall

**Review by eldenmoon — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**